tests/benchmark_overhead_works.rs

```diff
@@ -0,0 +1,69 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use assert_cmd::cargo::cargo_bin;
+use std::{process::Command, result::Result};
+use tempfile::tempdir;
+
+static RUNTIMES: [&'static str; 6] = ["polkadot", "kusama", "westend", "rococo", "wococo", "versi"];
+
+/// `benchmark-overhead` works for all dev runtimes.
+#[test]
+#[cfg(unix)]
+fn benchmark_overhead_works() {
+	for runtime in RUNTIMES {
+		assert!(benchmark_overhead(format!("{}-dev", runtime)).is_ok());
+	}
+}
+
+/// `benchmark-overhead` rejects all non-dev runtimes.
+#[test]
+#[cfg(unix)]
+fn benchmark_overhead_rejects_non_dev_runtimes() {
+	for runtime in RUNTIMES {
+		assert!(benchmark_overhead(runtime.into()).is_err());
+	}
+}
+
+fn benchmark_overhead(runtime: String) -> Result<(), String> {
+	let tmp_dir = tempdir().expect("could not create a temp dir");
+	let base_path = tmp_dir.path();
+
+	// Only put 5 extrinsics into the block otherwise it takes forever to build it
+	// especially for a non-release build.
+	let status = Command::new(cargo_bin("polkadot"))
+		.args(["benchmark-overhead", "--chain", &runtime])
+		.arg("-d")
+		.arg(base_path)
+		.arg("--weight-path")
+		.arg(base_path)
+		.args(["--warmup", "5", "--repeat", "5"])
+		.args(["--add", "100", "--mul", "1.2", "--metric", "p75"])
+		.args(["--max-ext-per-block", "5"])
+		.status()
+		.map_err(|e| format!("command failed: {:?}", e))?;
+
+	if !status.success() {
+		return Err("Command failed".into())
```

Comment:
Would be good to print stderr/stdout of the process to find out why it failed.

Reply:
The error output of the command is printed on failure, so it for example on an invalid argument:
```pre
error: Found argument '--unknown' which wasn't expected, or isn't valid in this context

        If you tried to supply `--unknown` as a value rather than a flag, use `-- --unknown`

USAGE:
    polkadot benchmark-overhead --chain <CHAIN_SPEC>

For more information try --help
```